src/utils/atomWithObservable.ts

```diff
@@ -29,69 +29,52 @@ type ObservableLike<T> = {
 
 type SubjectLike<T> = ObservableLike<T> & Observer<T>
 
+type InitialDataFunction<T> = () => T | undefined
+
+export type AtomWithObservableOptions<TData> = {
```

Comment:
I know it's a bit controversial, but we prefer not exporting types.

```suggestion
type AtomWithObservableOptions<TData> = {
```

Reply:
sure